crates/formatter/src/formatter.rs

```
@@ -28,16 +31,17 @@ impl Formatter {
 	pub fn format_root(self, root: &SyntaxNode) -> FormatResult {
 		let element = self
 			.format_syntax_node(root)
-			.unwrap_or_else(|| self.format_raw(root));
+			// TODO: we have an error here. Diagnostic?
```

Should the formatter ever be responsible for reporting syntax errors? I assume the parser would still emit diagnostics for every syntax error even if we recover and run other tools like the formatter.

Maybe this would be the difference between logging levels though, if you passed a `--log-level=verbose` it would also emit these kinds of events from other tools so we can debug why the formatter didn't format what we expected.

---

Yes it makes sense. Although executing an unwrap without a comment or way to handle it doesn't seem right, hence the `TODO` :)

---

I think we should either return an error or return the text as is